include/atomics/receiver_cadmium.hpp

```diff
@@ -0,0 +1,108 @@
+/**
+* Cristina Ruiz Martin
+* ARSLab - Carleton University
+*
+*/
+
+#ifndef __RECEIVER_CADMIUM_HPP__
+#define __RECEIVER_CADMIUM_HPP__
+
+#include <cadmium/modeling/ports.hpp>
+#include <cadmium/modeling/message_bag.hpp>
+#include <limits>
+#include <math.h> 
+#include <assert.h>
+#include <memory>
+#include <iomanip>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <chrono>
+#include <algorithm>
+#include <limits>
+#include <random>
+#include "../data_structures/message.hpp"
+
+using namespace cadmium;
+using namespace std;
+
+//Port definition
+struct Receiver_defs {
+	struct out : public out_port<Message_t> {
+    };
+    struct in : public in_port<Message_t> {
+    };
+};
+
+template<typename TIME>
+class Receiver {
+	using defs=Receiver_defs; // putting definitions in context
+	public:
+		//Parameters to be overwritten when instantiating the atomic model
+		TIME   PREPARATION_TIME;
+		// default constructor
+		Receiver() noexcept {
+			PREPARATION_TIME = TIME("00:00:10");
+			state.ack_num = 0;
+			state.sending = false;
+		}
+		// state definition
+		struct state_type {
+			int ack_num;
+			bool sending;
+		};
+        state_type state;
+        // ports definition
+        using input_ports = std::tuple<typename defs::in>;
+        using output_ports = std::tuple<typename defs::out>;
+        // internal transition
+        void internal_transition() {
+        	state.sending = false;
+        }
+        // external transition
+        void external_transition(TIME e,
+        		typename make_message_bags<input_ports>::type mbs) {
+        			if(get_messages<typename defs::in>(mbs).size()>1) {
+        				assert(false && "one message per time uniti");
+        			}
+        			for(const auto &x : get_messages<typename defs::in>(mbs)) {
+        				state.ack_num = static_cast < int > (x.value);
+        				state.sending = true;
+        			}
+                           
+        }
+        // confluence transition
+        void confluence_transition(TIME e,
+        		typename make_message_bags<input_ports>::type mbs) {
+        			internal_transition();
+        			external_transition(TIME(), std::move(mbs));
+        }
+        // output function
+        typename make_message_bags<output_ports>::type output() const {
+        	typename make_message_bags<output_ports>::type bags;
```

Comment:
Check for indentation please it should be 4 spaces.

Reply:
Please download the files to your local and check the indentations.
Its looking perfect in notepad++ or other editor with 4 spaces